python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -426,6 +430,7 @@ def __new__(
         asset_events: Optional[
             Sequence[Union[AssetObservation, AssetMaterialization, AssetCheckEvaluation]]
         ] = None,
+        **kwargs,
```

Comment:
we don't typically do this for whatever reason in NamedTuple constructors, not entirely sure why - certainly reduces boilerplate i suppose, but at the cost of typehinting / pyright catching stuff? if we don't plan to add a ton more going forward, maybe just bite the bullet and follow the cargo cult?

Reply:
My goal here was to avoid having this show up in typeahead for users who are working with the `SensorResult` object directly (which is probably at least somewhat rare, but still). It's that classic problem that Jamie also dealt with in backfill/RunRequest land of not wanting to have a non-public type for a parameter on a public class -- basically that `@hidden_param` thing that Nick wrote a treatise about a bit ago (but since that doesn't exist yet, I'm doing this the old-fashioned way)